sdk/storage/azure-storage-file-share/src/samples/java/com/azure/storage/file/share/ShareServiceAsyncJavaDocCodeSamples.java

```diff
@@ -286,4 +288,41 @@ public void generateAccountSas() {
         String sas = fileServiceAsyncClient.generateAccountSas(sasValues);
         // END: com.azure.storage.file.share.ShareServiceAsyncClient.generateAccountSas#AccountSasSignatureValues
     }
+
+    /**
+     * Code snippet for {@link ShareServiceAsyncClient#undeleteShare(String, String)}.
+     */
+    public void undeleteShare() {
+        ShareServiceAsyncClient fileServiceAsyncClient = createAsyncClientWithSASToken();
+        // BEGIN: com.azure.storage.file.share.ShareServiceAsyncClient.undeleteShare#String-String
+        ListSharesOptions listSharesOptions = new ListSharesOptions();
+        listSharesOptions.setIncludeDeleted(true);
+        fileServiceAsyncClient.listShares(listSharesOptions).flatMap(
```

Comment:
Are we guessing that the common pattern will be to list deleted shares to undelete them?

Reply:
Correct. We need two things to undelete a share - name and version. Listing API is the only way to obtain version.